cf-custom-resources/test/cert-replicator-test.js

```diff
@@ -0,0 +1,528 @@
+    "test.myapp.example.com",
+    "*.test.myapp.example.com",
+    "v1.myapp.example.com",
+    "v2.example.com",
+  ];
+  const testCopilotTags = [
+    { Key: "copilot-application", Value: testAppName },
+    { Key: "copilot-environment", Value: testEnvName },
+  ];
+  const testCertificateArn =
+    "arn:aws:acm:region:123456789012:certificate/12345678-1234-1234-1234-123456789012";
+  const spySleep = sinon.spy(function (ms) {
+    return Promise.resolve();
+  });
+
+  beforeEach(() => {
+    handler.withDefaultResponseURL(ResponseURL);
+    handler.withDefaultLogGroup(LogGroup);
+    handler.withDefaultLogStream(LogStream);
+    handler.withSleep(spySleep);
+    handler.withDeadlineExpired((_) => {
+      return new Promise(function (resolve, reject) {});
+    });
+    console.log = function () {};
+  });
+  afterEach(() => {
+    handler.reset();
+    AWS.restore();
+    console.log = origLog;
+    spySleep.resetHistory();
+  });
+
+  test("Empty event payload type fails", () => {
+    const request = nock(ResponseURL)
+      .put("/", (body) => {
+        return (
+          body.Status === "FAILED" &&
+          body.Reason ===
+            "Unsupported request type undefined (Log: /aws/lambda/testLambda/2021/06/28/[$LATEST]9b93a7dca7344adeb193d15c092dbbfd)"
+        );
+      })
+      .reply(200);
+    return LambdaTester(handler.certificateReplicateHandler)
+      .event({
+        RequestId: testRequestId,
+        ResourceProperties: {
+          AppName: testAppName,
+          EnvName: testEnvName,
+          TargetRegion: "us-east-1",
+          EnvRegion: "us-west-2",
+          CertificateArn: testCertificateArn,
+        },
+      })
+      .expectResolve(() => {
+        expect(request.isDone()).toBe(true);
+      });
+  });
+
+  test("Bogus operation fails", () => {
+    const bogusType = "bogus";
+    const request = nock(ResponseURL)
+      .put("/", (body) => {
+        return (
+          body.Status === "FAILED" &&
+          body.Reason ===
+            "Unsupported request type " +
+              bogusType +
+              " (Log: /aws/lambda/testLambda/2021/06/28/[$LATEST]9b93a7dca7344adeb193d15c092dbbfd)"
+        );
+      })
+      .reply(200);
+    return LambdaTester(handler.certificateReplicateHandler)
+      .event({
+        RequestType: bogusType,
+        RequestId: testRequestId,
+        ResourceProperties: {
+          AppName: testAppName,
+          EnvName: testEnvName,
+          TargetRegion: "us-east-1",
+          EnvRegion: "us-west-2",
+          CertificateArn: testCertificateArn,
+        },
+      })
+      .expectResolve(() => {
+        expect(request.isDone()).toBe(true);
+      });
+  });
+
+  test("Create operation fails if cannot describe old certificate", () => {
+    const describeCertificateFake = sinon.fake.rejects(new Error("some error"));
+
+    AWS.mock("ACM", "describeCertificate", describeCertificateFake);
+
+    const request = nock(ResponseURL)
+      .put("/", (body) => {
```

Comment:
Do you think we also want to match the `PhysicalResourceID` here? That it's a log stream name

Reply:
I updated my PR and we don't use log stream name for physical resource ID now